### DISCUSSION TITLE
Getting an error using action upload-artifact v3

### DISCUSSION THREAD

**Discussion by mwalter** [community/community#50642] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi,

after upgrading to upload-artifact v3 I'm getting this error breaking my build:

```
Run actions/upload-artifact@v3
With the provided path, there will be 5 files uploaded
Starting artifact upload
For more detailed logs during the artifact upload process, enable step-debugging: https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging
Error: Artifact name: , is incorrectly provided
```
I can't figure out why the artifact name is empty and why it is missing. Is there anybody who can tell me what I have to do to get the upload artifact action running? Thanks in advance!

Link to the broken build:
https://github.com/mwalter/MavenDependencyChecker/actions/runs/4483078376/jobs/7881923836

**Comment by airtower-luna** [comment, marked as answer]
The name is empty because the output you use as the value isn't set. This bit in the previous step is the problem:

```sh
          echo "name=filename::$FILENAME" >> $GITHUB_OUTPUT
```

That creates an output with the name `name`, and `filename::$FILENAME` (post variable substitution) as the value.

The following should work, assuming the value of `$FILENAME` is correct:

```sh
          echo "filename=$FILENAME" >> $GITHUB_OUTPUT
```
